tests/AMFDeserializeTests.cpp

```diff
@@ -0,0 +1,286 @@
+#include <chrono>
+#include <fstream>
+#include <iostream>
+
+#include "AMFDeserialize.h"
+#include "AMFFormat.h"
+#include "CommonCxxTests.h"
+
+AMFArrayValue* ReadFromBitStream(RakNet::BitStream* bitStream) {
+	AMFDeserialize reader;
+
+	return reader.Read(bitStream, true);
+}
+
+/**
+ * This test checks that if we recieve an unimplemented AMFValueType 
+ * we correctly throw an error and can actch it.
+ */
+void TestUnimplementedAMFValues() {
+	std::vector<AMFValueType> unimplementedValues = {
+		AMFValueType::AMFInteger,
+		AMFValueType::AMFXMLDoc,
+		AMFValueType::AMFDate,
+		AMFValueType::AMFObject,
+		AMFValueType::AMFXML,
+		AMFValueType::AMFByteArray,
+		AMFValueType::AMFVectorInt,
+		AMFValueType::AMFVectorUInt,
+		AMFValueType::AMFVectorDouble,
+		AMFValueType::AMFVectorObject,
+		AMFValueType::AMFDictionary
+		};
+	// Run unimplemented tests to check that errors are thrown if
+	// unimplemented AMF values are attempted to be parsed.
+	std::ifstream fileStream;
+	fileStream.open("AMFBitStreamUnimplementedTest.bin", std::ios::binary);
+
+	// Read a test BitStream from a file
+	std::vector<char> baseBitStream;
+	char byte = 0;
+	while (fileStream.get(byte)) {
+		baseBitStream.push_back(byte);
+	}
+
+	fileStream.close();
+
+	for (auto amfValueType : unimplementedValues) {
+		RakNet::BitStream testBitStream;
+		for (auto element : baseBitStream) {
+			testBitStream.Write(element);
+		}
+		testBitStream.Write(amfValueType);
+		bool caughtException = false;
+		try {
+			ReadFromBitStream(&testBitStream);
+		} catch (AMFValueType unimplementedValueType) {
+			caughtException = true;
+		}
+		std::cout << "Testing unimplemented value " << amfValueType << " Did we catch an exception: " << (caughtException ? "YES" : "NO") << std::endl;
+		assert(caughtException);
+	}
+}
+
+void TestLoad() {
```

Comment:
The goal of this test was to make sure the time it would take to parse an extraordinarily long packet would not exceed the maximum time we would want the server to spent parsing packets.  If the test is not useful it can be removed.